### PR TITLE
[Issue 1007] load caffe model from hdfs

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?

load caffe model from hdfs

## How was this patch tested?

unit test, integration test

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/1007

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 package com.intel.analytics.bigdl.utils
 
-import java.io.{FileInputStream, InputStreamReader}
+import java.io._
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala]: One of these 2 lines can be removed ?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala [hunk 2]
```diff
@@ -58,18 +60,41 @@ class CaffeLoader[T: ClassTag](prototxtPath: String, modelPath: String,
     }
   }
 
+  private def loadFromHdfs(fileName: String): FSDataInputStream = {
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala]: rename to another meaningful method name ?  this method just create the inputstream

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala (hunk 1 above)
Comment (SeaOfOcean) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala]: import java.io.{FileInputStream, InputStreamReader} is replaced by import java.io._

**Comment by SeaOfOcean** [comment]
pr validate and integration test pass

**Comment by wzhongyuan** [comment]
@SeaOfOcean  LGTM, please let me know when it's merged, then I will resolve the conflict and send the update

**Pull request merged**